File: snippets/typescript/scroll_points.ts
import { QdrantClient } from "@qdrant/js-client-rest";

const client = new QdrantClient({ host: "localhost", port: 6333 });

client.scroll("{collection_name}", {
    filter: {
        must: [
            {
                key: "country",
                match: {
                    value: "Canada",
                },
            },
        ],
    },
    limit: 1,
    with_payload: true,
    with_vector: false,
});

File: snippets/go/scroll_points.go
package client

import (
	"context"
	"fmt"

	"github.com/qdrant/go-client/qdrant"
)

func scroll() {
	client, err := qdrant.NewClient(&qdrant.Config{
		Host: "localhost",
		Port: 6334,
	})
	if err != nil {
		panic(err)
	}

	limit := uint32(1)
	results, err := client.Scroll(context.Background(), &qdrant.ScrollPoints{
		CollectionName: "{collection_name}",
		Filter: &qdrant.Filter{
			Must: []*qdrant.Condition{
				qdrant.NewMatch("color", "red"),
			},
		},
		Limit:       &limit,
		WithPayload: qdrant.NewWithPayloadEnable(true),
		WithVectors: qdrant.NewWithVectorsEnable(false),
	})
	if err != nil {
		panic(err)
	}
	fmt.Println("Results: ", results)
}
